File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {

        unordered_set <int> st;
        
        int maxNeg = INT_MIN;
        int sum = 0;

        for(int &n : nums){
            if(n<=0){
                maxNeg = max(n,maxNeg);
            }else if(!st.count(n)){
                sum+=n;
                st.insert(n);
            }
        }

        sum = sum == 0 ? maxNeg:sum;

        return sum;
    }
};

File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.java
class Solution {
    public int maxSum(int[] nums) {
        Set <Integer> set =new HashSet<>();

        int maxNeg = Integer.MIN_VALUE;

        int sum=0;

        for(int n : nums){
            if(n<=0){
                maxNeg = Math.max(maxNeg , n);
            }else if(!set.contains(n)){
                sum+=n;
                set.add(n);
            }
        }

        sum = (sum == 0)?maxNeg:sum;

        return sum;
    }
}